game-app/game-core/src/main/java/games/strategy/engine/framework/map/file/system/loader/ZippedMapsExtractor.java

```diff
@@ -72,10 +71,10 @@ public void unzipMapFiles() {
                 }));
   }
 
-  private Collection<File> findAllZippedMapFiles() {
-    return FileUtils.listFiles(downloadedMapsFolder.toFile()).stream()
-        .filter(File::isFile)
-        .filter(file -> file.getName().toLowerCase().endsWith(ZIP_EXTENSION))
+  private Collection<Path> findAllZippedMapFiles() {
+    return FileUtils.listFiles(downloadedMapsFolder).stream()
+        .filter(Files::isReadable)
```

Comment:
`isReadable` I don't think offhand is the same as 'isFile'. If not, then any directory that has a '.zip' extension would be returned (incorrectly). Please have a look into this to verify behavior.

Reply:
You're right. I was so sure that isReadable returned false for directories, but apparently it doesn't.
Good spot 👍🏼